### ISSUE TITLE
Plotting bug?

### ISSUE THREAD

**Issue by Fellow-baka** [SpinW/spinw#23]
In new version of matlab (9.5.0.1067069 (R2018b) Update 4 [previous updates too] and latest development branch of SpinW) it is not possible to select exchange bonds, atoms or any 'Patch' things (axes letters are ok) in plotted structure.
It seems the problem in patchfacefcn.m in barocentric coordinate calculation routine (near line 150).
I was not able to pinpoint what is going wrong, but dirty solution helps for now
```matlab
%   fIdx = pIdx(all(bCoord>=-tLim & bCoord<=1+tLim & repmat(sum(bCoord,2)<=1+tLim,1,2),2))
    fIdx = 1;
```
(Actually this selects only first bond/atom and isn't a solution)

**Comment by simonward86** [comment]
This is the same as issue #15 (please correct me if wrong). 
I also traced it to this point, however I found that this is not the problem. The problem seems to be in post R2017b, for some reason the mouse points have an offset which changes with extent and rotation. Hence, it doesn't find an intersect and `fIdx = []`.
Even more annoying is the fact that I can not create a minimal example for this behavior :-/  
Any further digging and suggestions are welcome as I'm a bit stuck

**Comment by Fellow-baka** [comment]
Yes, you are right, sorry for missing that.
I'll try to dig into it a bit more.

**Comment by Fellow-baka** [comment]
Found it. It seems there are additional transformation of hit.IntersectionPoint needed.
Could you reproduce this by just doubling this line in patchfacefcn.m?
``` matlab
P = (P-T)*(R)
P = (P-T)*(R)
```

**Comment by simonward86** [comment]
Great find! 
That was annoying me for some time. I've tested on R2017b, R2018a and R2018b and it seems to work. 
Fixed in commit 6e7c882 .
Nice one
